lib/msf/core/post/windows/services.rb

```diff
@@ -229,23 +260,40 @@ def service_list
   def service_info(name)
     service = {}
 
-    if load_extapi
-      begin
-        return session.extapi.service.query(name)
-      rescue Rex::Post::Meterpreter::RequestError => e
-        vprint_error("Request Error #{e} falling back to registry technique")
+    case session.type
+    when "powershell"
+      if SERVICES.empty?
```

Comment:
you could do something like this here:
```ruby
services = @services || list_services
```
and then instead of modifying a constant in `list_services`, use the instance variable `@services`